es/es-server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -219,17 +220,29 @@ protected void setupFieldType(BuilderContext context) {
                 fieldType.setHasDocValues(defaultDocValues);
             }
         }
+
+        public void position(int position) {
+            this.position = position;
+        }
     }
 
     protected final Version indexCreatedVersion;
     protected MappedFieldType fieldType;
     protected final MappedFieldType defaultFieldType;
     protected MultiFields multiFields;
     protected CopyTo copyTo;
-
-    protected FieldMapper(String simpleName, MappedFieldType fieldType, MappedFieldType defaultFieldType, Settings indexSettings, MultiFields multiFields, CopyTo copyTo) {
+    protected Integer position;
```

Comment:
If it is valid to pass in a `Null` for the `position` (like the PR is doing e.g. on IdMapper), we should annotate it imho (also the getter). also this would highlight why it isn't a primitive.